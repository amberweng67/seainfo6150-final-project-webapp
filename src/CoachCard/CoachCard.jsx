import React from "react";
import styles from './CoachCard.module.css';
import { Link } from "react-router-dom";

const CoachCard = ({ card }) => {
  return (
    <Link to={`/coaches/${card.id}`} className={styles.link}>
      <div className={styles.cardContainer} style={{ backgroundImage: `linear-gradient(transparent 36%, rgba(0, 0, 0, 0.87)), url(${require(`../assets/${card.id}.webp`)})` }}>
          <div>{card.name}</div>
          <div>{card.title}</div>
          <div>{card.company}</div>
      </div>
    </Link>
  );
};

export default CoachCard;