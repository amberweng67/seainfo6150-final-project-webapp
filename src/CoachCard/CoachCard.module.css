.cardContainer {
  height: 400px;
  width: 200px;
  background-size: cover;
  border-radius: 20px;
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  color: white;
  cursor: pointer;
}

.cardContainer > img {
  width: 100%;
}

.cardContainer:hover{
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.3);
  background-color: unset;
}

.link {
  border-radius: 20px;
  padding: unset;
}