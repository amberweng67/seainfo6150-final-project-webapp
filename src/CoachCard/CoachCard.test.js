import React from "react";
import CoachCard from "./CoachCard.jsx";

const CARD = {
  "id": "a45e2d9c-cc14-5d54-844f-c6da44f94d0f",
  "name": "Grace Lynch",
  "title": "Senior Product Manager",
  "company": "She inc",
  "intro": "Taught block rear previous pain though happily war wealth view image hay dull proud health eventually building food citizen necessary angry circus total middle"
};

it("renders the coachcard component correctly", () => {
  const { container } = render(<CoachCard card={CARD} />);
  expect(container).toMatchSnapshot();
});
