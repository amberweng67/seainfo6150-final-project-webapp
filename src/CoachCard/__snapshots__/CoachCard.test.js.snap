// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the coachcard component correctly 1`] = `
<div>
  <a
    class="link"
    href="/coaches/a45e2d9c-cc14-5d54-844f-c6da44f94d0f"
  >
    <div
      class="cardContainer"
    >
      <div>
        Grace Lynch
      </div>
      <div>
        Senior Product Manager
      </div>
      <div>
        She inc
      </div>
    </div>
  </a>
</div>
`;
